ddR/R/pdriver.R

```diff
@@ -1,90 +1,125 @@
 ###################################################################
 # Copyright 2015 Hewlett-Packard Development Company, L.P.
-# This program is free software; you can redistribute it 
-# and/or modify it under the terms of the GNU General Public 
+# This program is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public
 # License, version 2 as published by the Free Software Foundation.
 
-# This program is distributed in the hope that it will be useful, 
-# but WITHOUT ANY WARRANTY; without even the implied warranty of 
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU 
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 # General Public License for more details.
 
-# You should have received a copy of the GNU General Public License 
-# along with this program; if not, write to the Free Software 
-# Foundation, Inc., 59 Temple Place, Suite 330, 
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330,
 # Boston, MA 02111-1307 USA.
 ###################################################################
 
-setClass("ParallelddR", contains="ddRDriver")
-
-#' The default parallel driver
+#' @include ddR.R
+NULL
+
+setOldClass("SOCKcluster")
+setOldClass("cluster")
+setOldClass("spawnedMPIcluster")
+setClassUnion("parallelCluster", c("SOCKcluster", "cluster",
+                    "spawnedMPIcluster"))
+
+#' Class for parallel driver
+#'
+#' @slot type character "FORK" or "PSOCK"
+#' @slot cluster As returned from \link[parallel]{makeCluster}
+setClass("parallel.ddR", contains = "ddRDriver",
+        slots = c(type = "character", cluster = "parallelCluster"))
+
+
+windows <- (.Platform$OS.type == "windows")
+
+#' Initialize the no. of cores in parallel backend
+#'
+#' The FORK method of parallel works only on UNIX environments. The "PSOCK"
+#' method requires SNOW but works on all OSes.
+#'
+#' This function is a wrapper around \link[parallel]{makeCluster}
+#'
+#' @param spec Number of cores to run with, or character vector of
+#'      hostnames for SNOW cluster. If NULL defaults to the number of
+#'      physical cores on machine.
+#' @param type If "FORK", will use UNIX fork(). If "PSOCK", will use SNOW.
+#' @param ... Additional arguments to \link[parallel]{makeCluster}
+#' @return Object of class \code{\linkS4class{parallel.ddR}} representing a running parallel
+#'      cluster
+#'
 #' @examples
 #' \dontrun{
-#' useBackend(parallel,executors=4)
+#' # Cluster on 2 cores with default type:
+#' useBackend("parallel", 2)
+#'
+#' # PSOCK cluster on multiple machines (assuming you can ssh user@server1):
+#' servers <- c("server1", "server2")
+#' useBackend("parallel", servers, type = "PSOCK")
```

Comment:
Machine names can be replicated, e.g. rep(c("server1", "server2"), each=4)
to get 8 workers in total.

On Sep 4, 2016 22:43, "Dmitriy Selivanov" <notifications@github.com> wrote:
>
> In ddR/R/pdriver.R:
>
> > #' @examples > #' \dontrun{ > -#' useBackend(parallel,executors=4) >
+#' # Cluster on 2 cores with default type: > +#' useBackend("parallel", 2)
> +#' > +#' # PSOCK cluster on multiple machines (assuming you can ssh
user@server1): > +#' servers <- c("server1", "server2") > +#'
useBackend("parallel", servers, type = "PSOCK")
>
> Imagine I have 2 machines with 4 cores on each. How to create "PSOCK"
cluster with 8 workers?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.